### PR TITLE
Milestone 5 Complete

### PR DESCRIPTION
**Pull request by ivanid22**
I've just completed milestone 5 of the React calculator project. In this assignment, I've connected the calculator logic to the React App component. I've had to refactor the calculate method a bit to be able to keep track of the application state independently. 
Looking forward to your feedback!

Thanks!

### CODE REVIEW

File: src/components/ButtonPanel.js [hunk 1]
```diff
@@ -1,12 +1,9 @@
...
+import PropTypes from 'prop-types';
 import { v4 as genKey } from 'uuid';
 import Button from './Button';
 
-const renderGroup = group => (
-  group.map(item => <Button name={item} key={genKey()} wide={item === '0'} />)
-);
-
-const ButtonPanel = () => {
+const ButtonPanel = props => {
```
Comment (Forison) [inline src/components/ButtonPanel.js]: [Optional]
You could destructure clickhandler at the argument level `const ButtonPanel = ({clickhandler}) => {`

File: src/components/App.js [hunk 2]
```diff
@@ -1,14 +1,102 @@
...
+      currentlyDisplayedValue: state.next ? state.next + val : val,
+    }));
+  }
+
+  handleClick(buttonName) {
+    const {
+      total, next, operation, justCalculated,
+    } = this.state;
+
+    if (DIGITS.includes(buttonName)) {
```
Comment (Forison) [inline src/components/App.js]: * [x] Kindly move this logic to calculate.js

**Review by Forison — CHANGES_REQUESTED** [review]
## Status:: Changes required❌.
# 🙋‍♂️ @ivanid22 
#### You have done a good job on this one, I like the way the display could accommodate numbers without the number overflowing. Your calculator is almost as required. Let us address these few issues to get the project approved🤗.
* [x] The project has a descriptive README.md ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_readme.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
- Kindly add an image of the application in the readme.
- The readme should include how to set up and use the project on ones's local machine.
* [x] When a number is divided by zero the application breaks, it ought to return `undefined` instead.
* [x] When `%` is selected on a single number ie 10 the application should return 0.1
* [x] When `%` is selected on an operation ie `10 +20` the application should return 0.3
- this way we can perform further computation and have a more robust calculator.
* [x] Check comment below in relation to moving code into `calculate.js` 
**[OPTIONAL]**
* [x] Kindly add a description to all repository you create on Github.
* [x] The application can have a better user experience if the selected operation and the next value are all visible as the user performs a computation.
### That is kindly address the issue and submit a re-review👨‍🔧.
#### Happy coding👨‍💻
Addo Forison

**Review by whiz25 — APPROVED** [review]
## Approved :clap: 
You have done a good job fixing the issues addressed by the previous reviewer. Merge and proceed to the next project.


### Martin Afani

**Pull request merged**